srtp/srtp.c

```diff
@@ -1482,6 +1786,16 @@ srtp_unprotect_aead (srtp_ctx_t *ctx, srtp_stream_ctx_t *stream, int delta,
     }
   }
 
+  if (xtn_hdr && stream->rtp_xtn_hdr_cipher) {
+    /*
+     * extensions header encryption RFC 6904
+     */
+    status = srtp_process_header_encryption(stream, xtn_hdr);
```

Comment:
Be aware that you will be decrypting data here prior to any authentication of the data.  With GCM mode, the authentication occurs when decrypting the payload, which hasn't occurred yet.  IMHO, this is a criticism of AEAD ciphers.  The authentication occurs after decryption.  You may want to consider decrypting the header extensions *after* the payload has been decrypted and authenticated.